src/bun.js/api/server.zig

```diff
@@ -2106,7 +2106,7 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
 
             this.setAbortHandler();
             streamLog("is in progress, but did not return a Promise. Finalizing request context", .{});
-            this.finalize();
+            this.finalizeWithoutDeinit();
```

Comment:
Can you check if this leaks the memory?

It's possible we should create a JSCell for RequestContext and then just rely on GC to manage the lifetime of the request instead of trying to be smart and cover all the edgecases

Reply:
ill definitely check for leaks from this. but we cannot free it at this point because onAbort can still get called.